### DISCUSSION TITLE
Loss of data when using the fullscreen mode

### DISCUSSION THREAD

**Discussion by anelleh** [jspsych/jsPsych#1645] [Q&A]
Hi! 

We're currently piloting a jspsych experiment that is hosted on Cognition.run. Unfortunately we've experienced an issue after implementing the fullscreen mode with the jspsych fullscreen plugin, which is that we lost some of the data for two of the participants (only two last trials for one participant, but twenty for the other participant). After asking these two participants if they experienced any technical difficulties during the experiment, both reported that their computers went briefly into screensaver mode, but that they were able to complete the experiment and didn't feel like they missed any of the trials. As we're directing the participants to a short post-session feedback questionnaire at the end of the experiment via an external link and both of these participants completed the questionnaire, we know that they completed the experiment.

I couldn't find anything related to this in the discussion forum, so I was wondering if this issue has been reported before and if there's anything we can do about it? 
Thanks in advance if anyone can help!

**Comment by jodeleeuw** [comment]
Hi @anneleh! I haven't heard of a problem like this before. Can you say more about the task? I'm especially curious about how the experiment could go into screensaver mode if they are interacting with the page.

**Reply by anelleh** [reply]
Hi @jodeleeuw,

Thanks a lot for the quick reply! We're actually collecting audio data with the new html-audio-response plugin, so the participants aren't really interacting with the page - they only see strings of letters and give a spoken response according to the instructions. The whole experiment consists of four tasks: the first one consists of 30 consecutive html-audio-response trials, the second 72, the third one 30 and the last one (where the experiment has gone into screensaver mode for two of the participants) again of 72 trials. The whole duration of the last task is the same as that of the second task (less than 5 minutes). Of course the participants move on from one task to the other with a button press, so they are interacting with the page during the experiment.

**Reply by jodeleeuw** [reply]
I have very little experience with the various audio-response plugins that have been created. @becky-gilbert is the expert here, so I'm tagging her to see if she's ever encountered a similar issue with screens going to sleep or data loss.

**Comment by becky-gilbert** [comment]
Hi @anelleh, I'm afraid I've never had this happen or heard of this happening before. I agree with Josh that it seems strange that the computer would go into screensaver mode during a task. Perhaps this is related to the participants' computer settings, and whether or not they were using a laptop that had low battery and/or wasn't plugged in (in which case the screen may turn off more quickly). You could try testing this yourself by changing your computer settings so that the screensaver turns on after a very short period (less than your task duration) and see if you can replicate the problem. Here's what these settings look like on my Windows 10 laptop:
![screen_sleep](https://user-images.githubusercontent.com/9041788/112229682-a7e2c280-8bf0-11eb-988d-32d3dcc15853.PNG)

And maybe a workaround is to add some trials to your task that just say something like 'please click next to continue' with a next button. If these trials appear every 1-2 minutes, then hopefully that would prevent the computer screen from turning off due to inactivity. 

Sorry that I don't have more insight into this. Please let us know if you have any updates that might help us figure out the problem.

**Reply by anelleh** [reply]
Hi @becky-gilbert,

Thank you for the suggestion! I was kind of able to replicate the problem by changing my computer settings as you suggested (though only data for the last few trials was lost), but, interestingly, when I clicked on the experiment link again, I got the notification "Previously stored data has been sent to the server", after which all the data for the previous experiment was saved. I imagine that if the two participants had clicked on the experiment link again after completing the experiment, all of their data may have been saved as well (we could ask them to do it now, but I'm not sure if this would work after the experiment has been completed a week ago).

I'm not sure if we can add button-press trials to the tasks, but we'll try to see if there's anything we can do about this problem. We're planning on collecting the actual data at a computer lab at the university where we have full control over the computer settings, so this shouldn't be a major issue for us, but in case we need to collect the data remotely we may have to address this issue somehow.
